app/src/main/java/com/gh4a/fragment/ReleaseListFragment.java

```diff
@@ -39,9 +48,16 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
 
     @Override
     protected Single<Response<Page<Release>>> loadPage(int page, boolean bypassCache) {
-        final RepositoryReleaseService service =
-                ServiceFactory.get(RepositoryReleaseService.class, bypassCache);
-        return service.getReleases(mUserLogin, mRepoName, page);
+        final RepositoryReleaseService service = ServiceFactory.get(RepositoryReleaseService.class, bypassCache);
+        return service.getReleases(mUserLogin, mRepoName, page)
+                // Sometimes the API returns releases in a slightly wrong order (see TeamNewPipe/NewPipe repo
+                // for an example), so we need to fix the sorting locally
+                .map(response -> {
+                    if (response.body() != null) {
+                        Collections.sort(response.body().items(), MOST_RECENT_RELEASES_AND_DRAFTS_FIRST);
```

Comment:
This only works if the misordering doesn't happen across pages. Not sure whether this assumption is safe...
Maybe asking GH support would be worth a try.

Reply:
> This only works if the misordering doesn't happen across pages.

I'm aware of that, and I think this is good enough. In that case, the specific misordering that is across pages won't be solved by the workaround. I think it can be acceptable, given that as far as I've seen the sorting is only slightly wrong (off by 1 or 2 at worst).

Reply:
Btw, did you use https://support.github.com to ask them in the past?

Reply:
I used to send them emails, but lately the only way of contacting them is the community forum (https://github.community).

Reply:
I've sent them a support ticket, let's see what they say.

Reply:
@maniac103 Update on the situation: together with GH support, I discovered that this "almost correct" sorting behavior happens also when using the GraphQL without specifying an `orderBy` clause.
Therefore we concluded that this isn't really a bug, but the default behavior of the releases list when no sorting criteria is specified.
I asked them to consider changing the default sorting for the releases REST API, but I don't know if and when they'll do it. 
At this point we have three alternatives:
- wait to see whether they accept to change the releases REST API (not much likely imo)
- reimplement the releases list using the GraphQL API, since that one allows to specify the sorting criteria
- keep this limited but acceptable workaround that does its work

What do you think?